### ISSUE TITLE
Start sync / copy without enumerating host container

### ISSUE THREAD

**Issue by mforde84** [rclone/rclone#3915]
#### What is your current rclone version (output from `rclone version`)?

rclone v1.50.2
- os/arch: linux/amd64
- go version: go1.13.4

#### What problem are you are trying to solve?

I'm trying to improve speed when copying / syncing a large swift object container to another target swift cluster. It's a basic migration project. Our containers tend to be quite large, e.g., 1-5M objects. So to improve throughout I do the following:

1) enumerate the entire object list of the container
    $ swift list some.container > container.list
2) split the object list into 10000 object bins
    $ split -l 10000 container.list container.splits.
3) run rclone in parallel using each bin as a filter
    for parallel, we use slurm cluster with the following workload:
    $ rclone copy --use-mmap --buffer-size 99M 'HOST':'CONTAINER' 'TARGET':'CONTAINER' --files-from 'CONTAINER.SPLIT' --log-file='LOG LOCATION' --stats 1m --stats-one-line-date --size-only --checkers 500 --transfers 500
                 
So this works pretty well for smaller containers. But once we start doing large containers, we run into an issue when rclone attempts to enumerate the container.  If I understand the process right, rclone first enumerates the full object list for 'HOST':'CONTAINER', then greps / filters this list with the object list provided by the --file-from flag. 

This adds a significant amount of time to the transfer process, as it will typically take ~5-10 minutes to enumerate the container and for transfers to actually initiate. Also this process is repeated if there are any resulting failures following completion of a full sync attempt. 

So it seems like rclone is being bogged down here at multiple points in our process just attempting to enumerate the host container. In our scenario, this seems largely unnecessary since we already know the files we want to transfer. Is there a way to force rclone to skip this enumeration, and take the filter file as "given". If given in this sense, shouldn't rclone be able to assume the files within the provided manifest exist, and just start transferring them.

#### How do you think rclone should be changed to solve that?

Would be great if we could force rclone to ignore enumeration of a container when used in conjunction with a filter file. Also would be nice to skip the enumeration step if the whole sync fails. For example, if 10 files out of 10,000 files in the filter file fail, to just retry those 10, as opposed to enumerating the entire container, filtering that list, then checking all 10,000 files.

**Comment by ncw** [comment]
>  If I understand the process right, rclone first enumerates the full object list for 'HOST':'CONTAINER', then greps / filters this list with the object list provided by the --file-from flag.

That is the default, the alternative is to add --no-traverse which when used with --files-from stops that enumerations. As part of building the `--files-from` memory structure, rclone will HEAD each object. It does this in parallel (with checkers)  so may/may not be quicker.

If you are using [the latest beta](https://beta.rclone.org) you can also use

      --no-check-dest          Don't check the destination, copy regardless.

which will re-copy stuff if you use it carelessly but can improve performance.

Did you try doing the whole transfer with rclone? I think that will give better performance than with using a --files-from list especially if you use --fast-list.

> For example, if 10 files out of 10,000 files in the filter file fail, to just retry those 10, as opposed to enumerating the entire container, filtering that list, then checking all 10,000 files.

This probably deserves its own issue. It is a good idea and I'm not sure why we haven't implemented it before. Some errors you need to retry (like a listing failure) but if you got a specific error on a specific file no reason you couldn't just retry that file...

**Comment by animosity22** [comment]
I think this looks answered. Please let us know if there is still a question here or it can be closed out.

**Comment by mforde84** [comment]
It can be closed. Thanks for your help.
